evalml/model_understanding/graphs.py

```diff
@@ -569,6 +572,90 @@ def graph_prediction_vs_actual(y_true, y_pred, outlier_threshold=None):
     return _go.Figure(layout=layout, data=data)
 
 
+def get_tree_data(clf):
+    """Return data for a fitted tree in a restructured format
+
+    Arguments:
+        clf (ComponentBase): A fitted tree-based estimator.
+
+    Returns:
+        OrderedDict: An OrderedDict of OrderedDicts describing a tree structure
+    """
+    if not clf.model_family == ModelFamily.DECISION_TREE:
+        raise ValueError("Tree structure reformatting is not supported for non-Tree estimators")
+    est = clf._component_obj
+
+    children_left = est.tree_.children_left
+    children_right = est.tree_.children_right
+    features = est.tree_.feature
+    thresholds = est.tree_.threshold
+    values = est.tree_.value
+
+    def recurse(i):
+        if children_left[i] == children_right[i]:
+            return {'Name': f'Leaf_{i}'}
+        return OrderedDict({
+            'Feature': features[i],
+            'Threshold': thresholds[i],
+            'Value': values[i],
+            'Left_Child': recurse(children_left[i]),
+            'Right_Child': recurse(children_right[i])
+        })
+
+    return recurse(0)
+
+
+def visualize_decision_tree(clf, max_depth=None, rotate=False, filled=False, filepath=None):
+    """Generate an image visualizing the decision tree
+
+    Arguments:
+        clf (ComponentBase): A fitted tree-based estimator.
+        max_depth (int, optional): The depth to which the tree should be displayed.
+        rotate (bool, optional): Orient tree left to right rather than top-down.
+        filled (bool, optional): Paint nodes to indicate majority class for classification, extremity of values for
+        regression, or purity of node for multi-output.
+        filepath (str, optional): Path to where the graph should be saved. If set to None (as by default), the graph
```

Comment:
@ParthivNaresh pretty cool that you added the ability to save output to file!

I just filed #1535 to track coming up with a uniform way to save graphs/plots, for all our model understanding tools. I think its fine to keep this `filepath` arg here for the time being; I think the ultimate goal is to have one pattern across all the model understanding tools.